packages/sdk/src/sign.ts

```diff
@@ -217,6 +231,27 @@ export const unpackEnvelope = async (
     return makeParsedEvent(event, envelope.hash, envelope.signature)
 }
 
+export const unpackSnapshot = async (
+    snapshot: Envelope,
+    opts: UnpackEnvelopeOpts | undefined,
+): Promise<ParsedSnapshot> => {
+    check(hasElements(snapshot.event), 'Snapshot base is not set', Err.BAD_EVENT)
+    check(hasElements(snapshot.hash), 'Snapshot hash is not set', Err.BAD_EVENT)
+    check(hasElements(snapshot.signature), 'Snapshot signature is not set', Err.BAD_EVENT)
+
+    const sn = fromBinary(SnapshotSchema, snapshot.event)
+    let hash = snapshot.hash
+
+    if (opts?.disableHashValidation !== true) {
+        hash = riverSnapshotHash(snapshot.event)
+        check(bin_equal(hash, snapshot.hash), 'Snapshot id is not valid', Err.BAD_EVENT_ID)
+    }
+
+    // TODO: Check signature?
```

Comment:
in order to check the signature we need the header that the snapshot was originally linked to... is that accessible from this point in the code? can we pass it to this function?